src/prefect/filesystems.py

```diff
@@ -791,47 +793,103 @@ class GitHub(ReadableDeploymentStorage):
         default=None,
         description="An optional reference to pin to; can be a branch name or tag.",
     )
+    access_token: Optional[SecretStr] = Field(
+        name="Personal Access Token",
+        default=None,
+        description="A github Personal Access Token (PAT).",
+    )
+
+    @validator("access_token")
```

Comment:
Could be useful to throw a comment above this about why you made this an `access_token` validator instead of a `repository` validator (I think I get it - you only want to validate when a PAT is provided)